clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -2503,7 +2543,20 @@ void CXXNameMangler::mangleType(QualType T) {
   const Type *ty = split.Ty;
 
   bool isSubstitutable =
-    isTypeSubstitutable(quals, ty, Context.getASTContext());
+      isTypeSubstitutable(quals, ty, Context.getASTContext());
+  if (Context.isUniqueNameMangler()) {
+    static std::array<DeclContextDesc, 5> Scopes = {
```

Comment:
This can be const instead of static I think.  Additionally, I think that this functionality should just be its own function (isSyclHalfType) so that we only have to do this once.

Reply:
Added `const` as well. I think `static` makes sense because these scopes are the always the same